apps/web/src/components/Composer/Actions/SpaceSettings/TokenGateForm.tsx

```diff
@@ -0,0 +1,173 @@
+import MenuTransition from '@components/Shared/MenuTransition';
+import Search from '@components/Shared/Navbar/Search';
+import { Menu } from '@headlessui/react';
+import { ChevronDownIcon } from '@heroicons/react/outline';
+import { CheckCircleIcon } from '@heroicons/react/solid';
+import type { Profile } from '@lenster/lens';
+import { Input, Toggle } from '@lenster/ui';
+import { t, Trans } from '@lingui/macro';
+import clsx from 'clsx';
+import type { Dispatch, FC, SetStateAction } from 'react';
+import React from 'react';
+import { TokenGateCondition } from 'src/enums';
+import { useSpacesStore } from 'src/store/spaces';
+
+interface TokenGateFormProps {
+  setShowModal: Dispatch<SetStateAction<boolean>>;
+}
+
+const TokenGateForm: FC<TokenGateFormProps> = ({ setShowModal }) => {
+  const {
+    isTokenGated,
+    setIsTokenGated,
+    setTokenGateConditionType,
+    tokenGateConditionType,
+    setTokenGateConditionValue,
+    tokenGateConditionValue
+  } = useSpacesStore();
+
+  interface ModuleProps {
+    title: string;
+    onClick: () => void;
+    condition: TokenGateCondition;
+  }
+
+  const getTokenGateConditionDescription = (
+    tokenGateConditionType: TokenGateCondition
+  ) => {
+    switch (tokenGateConditionType) {
+      case TokenGateCondition.HAVE_A_LENS_PROFILE:
+        return t`have a lens profile`;
+      case TokenGateCondition.FOLLOW_A_LENS_PROFILE:
+        return t`follow a lens profile`;
+      case TokenGateCondition.COLLECT_A_POST:
+        return t`collect a post`;
+      case TokenGateCondition.MIRROR_A_POST:
+        return t`mirror a post`;
+    }
+  };
+
+  const onProfileSelected = (profile: Profile) => {
+    setTokenGateConditionValue(profile.handle);
+    setShowModal(false);
+  };
+
+  const Module: FC<ModuleProps> = ({ title, onClick, condition }) => (
+    <Menu.Item
+      as="button"
+      className={clsx(
+        { 'dropdown-active': tokenGateConditionType === condition },
+        'menu-item w-[11rem]'
```

Comment:
```suggestion
        'menu-item w-44'
```
I guess 44 is the correct value. Don't use arbitrary values

Reply:
Updated